### PR TITLE
Documentation on Project-Urls

### PR DESCRIPTION
**Pull request by DarkaMaul**
Closes https://github.com/pypi/warehouse/issues/16486

/cc @woodruffw @facutuesca

### CODE REVIEW

File: docs/user/project_metadata.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+PyPI renders these URLs on the package project page. They are also available in 
+the JSON data when requesting the package information.
+
+### Verified details
+
+![Verified details](assets/verified_details.png){ loading=lazy }
+
+
+[Trusted Publishing](trusted-publishers) allow PyPI to verify certain
```
Comment (facutuesca) [inline docs/user/project_metadata.md:17]:
```suggestion
[Trusted Publishing](trusted-publishers) allows PyPI to verify certain 
```

File: docs/user/project_metadata.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+[Trusted Publishing](trusted-publishers) allow PyPI to verify certain 
+information submitted by the package maintainer. Verified information appears in
+a separate section of the package detail page. A verified URL will have a green 
+checkmark attached (:fontawesome-solid-circle-check:{ .checked }).
+
+!!! warning
+
+    An URL being verified only attest that the _provenance_ of the build has 
+    been verified, i.e. the packages sources are coming from the URL specified.
```
Comment (facutuesca) [inline docs/user/project_metadata.md:25]: > the package sources are coming from the URL specified

This is not always the case: for example, we mark `pypi.org/p/$PROJECT_NAME` as verified. We are also planning to add support for verifying ReadTheDocs URLs. I would maybe remove this for now, or add a placeholder so that when we have a document specifying how/which URLs are verified, we can link to that.

File: docs/user/project_metadata.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    PyPI does not check the package content itself.
+
+
+### Icons
+
+![Unverified details](assets/unverified_details.png){ loading=lazy }
+
+While the labels can be arbitrary, `warehouse` recognizes the one from the
```
Comment (facutuesca) [inline docs/user/project_metadata.md:34]:
```suggestion
While the labels can be arbitrary, `warehouse` recognizes the ones from the
```

File: docs/user/project_metadata.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+| :fontawesome-solid-bug:           | Bug | Bug/Issue report location | :fontawesome-solid-info:{ title="Issue, Tracker, Report"} |
+| :fontawesome-solid-circle-dollar-to-slot: | Funding | Sponsoring information | :fontawesome-solid-info:{ title="Sponsor, Donation, Donate"}                               |
+
+
+#### Hosting Platforms
+
+| Icon                           | Name      |
+|:-------------------------------|:----------|
+| :fontawesome-brands-github:    | GitHub    |
+| :fontawesome-brands-gitlab:    | Gitlab    |
```
Comment (facutuesca) [inline docs/user/project_metadata.md:55]:
```suggestion
| :fontawesome-brands-gitlab:    | GitLab    |
```

File: warehouse/templates/packaging/detail.html [hunk 5]
```diff
@@ -45,9 +45,9 @@
   {% set icon = "fab fa-reddit-alien" %}
 {% elif name.lower().startswith("slack") or parsed.netloc == "slack.com" or parsed.netloc.endswith(".slack.com") %}
   {% set icon = "fab fa-slack" %}
-{% elif parsed.netloc == "twitter.com" or parsed.netloc.endswith(".twitter.com") %}
+{% elif parsed.netloc in ["twitter.com", "x.com"] or parsed.netloc.endswith((".twitter.com", "x.com")) %}
```
Comment (facutuesca) [inline warehouse/templates/packaging/detail.html:48]:
```suggestion
{% elif parsed.netloc in ["twitter.com", "x.com"] or parsed.netloc.endswith((".twitter.com", ".x.com")) %}
```

File: docs/user/project_metadata.md [hunk 6]
```diff
@@ -0,0 +1,88 @@
+# Projects metadata
```
Comment (di) [inline docs/user/project_metadata.md:1]:
```suggestion
# Project metadata
```

File: docs/user/project_metadata.md [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+
+## Project-URL
+
+A package can contain optional [Project URL](https://packaging.python.org/en/latest/specifications/core-metadata/#project-url-multiple-use)
```
Comment (di) [inline docs/user/project_metadata.md:6]:
```suggestion
Package metadata can contain optional [`Project-URL`](https://packaging.python.org/en/latest/specifications/core-metadata/#project-url-multiple-use) 
```

File: docs/user/project_metadata.md [hunk 8]
```diff
@@ -0,0 +1,88 @@
+# Projects metadata
+
+
+## Project-URL
+
+A package can contain optional [Project URL](https://packaging.python.org/en/latest/specifications/core-metadata/#project-url-multiple-use) 
+keys.
+
+PyPI renders these URLs on the package project page. They are also available in
```
Comment (di) [inline docs/user/project_metadata.md:9]:
```suggestion
PyPI renders these URLs on the project page. They are also available in 
```

File: docs/user/project_metadata.md [hunk 9]
```diff
@@ -0,0 +1,88 @@
+# Projects metadata
+
+
+## Project-URL
+
+A package can contain optional [Project URL](https://packaging.python.org/en/latest/specifications/core-metadata/#project-url-multiple-use) 
+keys.
+
+PyPI renders these URLs on the package project page. They are also available in 
+the JSON data when requesting the package information.
```
Comment (di) [inline docs/user/project_metadata.md:10]:
```suggestion
the [JSON API](https://warehouse.pypa.io/api-reference/json.html).
```

File: docs/user/project_metadata.md [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+
+### Verified details
+
+![Verified details](assets/verified_details.png){ loading=lazy }
+
+
```
Comment (di) [inline docs/user/project_metadata.md:16]:
```suggestion
PyPI currently supports several ways of verifying project URLs:

#### Self-links
Any URL to the project on PyPI will be considered verified.

#### Via Trusted Publishing
```
I'd like to make this less Trusted Publishing focused.

File: docs/user/project_metadata.md [hunk 11]
```diff
@@ -0,0 +1,88 @@
...
+[Trusted Publishing](trusted-publishers) allow PyPI to verify certain 
+information submitted by the package maintainer. Verified information appears in
```
Comment (di) [inline docs/user/project_metadata.md:18]: Let's be clear about what information we use for verification here -- username and repository name. We should also be clear about which providers we support this for and which we don't.

File: docs/user/project_metadata.md [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+a separate section of the package detail page. A verified URL will have a green 
+checkmark attached (:fontawesome-solid-circle-check:{ .checked }).
+
+!!! warning
+
+    An URL being verified only attest that the _provenance_ of the build has 
+    been verified, i.e. the packages sources are coming from the URL specified.
+
+    PyPI does not check the package content itself.
```
Comment (di) [inline docs/user/project_metadata.md:27]: I would rephrase this entirely:

```suggestion
    An URL being verified only attests that the URL is under control of the
    PyPI package owner, and does not imply any additional safety about that
    URL or any other relationship to the project in question.
```

File: docs/user/project_metadata.md (hunk 3 above)
Comment (di) [inline docs/user/project_metadata.md:34]:
```suggestion
While the labels can be arbitrary, PyPI recognizes the ones from the
```

Since this is user-facing documentation, we shouldn't use the codename 'warehouse' here.

File: docs/user/project_metadata.md (hunk 11 above)
Comment (di) [reply docs/user/project_metadata.md]: I think this still needs to be resolved.

File: docs/user/project_metadata.md [hunk 13]
```diff
@@ -0,0 +1,106 @@
+# Project Metadata
+
+
+## Project-URL
```
Comment (woodruffw) [inline docs/user/project_metadata.md:4]: Nitpick: `Project-URL` is the metadata name at distribution bake time, but most users probably know it better as the `[project.urls]` table. I think we should make the title here just "Project URLs" and link to the appropriate part of the `pyproject.toml` spec, rather than the core metadata spec.

```suggestion
## Project URLs
```

File: docs/user/project_metadata.md [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+[gh-action-tab]: trusted-publishers/creating-a-project-through-oidc.md#github-actions
+[gc-tab]: trusted-publishers/creating-a-project-through-oidc.md#google-cloud
+[active-tab]: trusted-publishers/creating-a-project-through-oidc.md#activestate
+[gitlab-tab]: trusted-publishers/creating-a-project-through-oidc.md#gitlab-cicd
+
+!!! warning
+
+    An URL being verified only attests that the URL is under control of the
+    PyPI package owner, and does not imply any additional safety about that
+    URL or any other relationship to the project in question.
```
Comment (di) [inline docs/user/project_metadata.md:46]: I think this can get moved...
```suggestion
```

File: docs/user/project_metadata.md [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+PyPI renders these URLs on the project page. They are also available in 
+the [JSON API](https://warehouse.pypa.io/api-reference/json.html).
+
+### Verified details
+
+![Verified details](assets/verified_details.png){ loading=lazy }
+
+PyPI currently supports several ways of verifying project URLs and displays 
+marks them using a green checkmark (:fontawesome-solid-circle-check:{ .checked }).
+
```
Comment (di) [inline docs/user/project_metadata.md:19]: ...here.
```suggestion

!!! warning

    An URL being verified only attests that the URL is under control of the
    PyPI package owner, and does not imply any additional safety about that
    URL or any other relationship to the project in question.

```

File: docs/user/project_metadata.md [hunk 16]
```diff
@@ -0,0 +1,106 @@
+# Project Metadata
+
+
+## Project-URL
+
+Package metadata can contain optional [`Project-URL`](https://packaging.python.
+org/en/latest/specifications/core-metadata/#project-url-multiple-use) 
+keys.
```
Comment (woodruffw) [inline docs/user/project_metadata.md:8]:
```suggestion
Packages can specify URLs in their metadata via the [`[project.urls]` table] in the package's `pyproject.toml`.

[`[project.urls] table`]: https://packaging.python.org/en/latest/specifications/pyproject-toml/#urls
```

File: docs/user/project_metadata.md [hunk 17]
```diff
@@ -0,0 +1,106 @@
...
+|:------------------------------------------|:--------------|:--------------------------|:-------------------------------------------------------------|
+| :fontawesome-solid-house-chimney:         | Homepage      | For the project homepage  |                                                              |
+| :fontawesome-solid-cloud:                 | Download      | A download link           |                                                              |
+| :fontawesome-solid-scroll:                | Changelog     | Changelog information     |                                                              |
+| :fontawesome-solid-book:                  | Documentation | Project documentation     | :fontawesome-solid-info:{ title="Docs" }                     |
+| :fontawesome-solid-bug:                   | Bug           | Bug/Issue report location | :fontawesome-solid-info:{ title="Issue, Tracker, Report"}    |
+| :fontawesome-solid-circle-dollar-to-slot: | Funding       | Sponsoring information    | :fontawesome-solid-info:{ title="Sponsor, Donation, Donate"} |
```
Comment (di) [inline docs/user/project_metadata.md:65]: I think including the "info" icon with a tooltip is confusing here, let's just include the text in the cell instead.

**Review by di — APPROVED** [review]

**Comment by di** [comment]
@woodruffw Want to give a final approval here?

File: docs/mkdocs-user-docs.yml [hunk 18]
```diff
@@ -10,19 +10,25 @@ plugins:
       j2_variable_end_string: "]]-->"
 markdown_extensions:
   - admonition
+  - attr_list
+  - footnotes
```
Comment (woodruffw) [inline docs/mkdocs-user-docs.yml:14]: Nitpick: I believe there are no more footnotes on these pages, so this _could_ be dropped?

```suggestion
```

File: docs/user/project_metadata.md [hunk 19]
```diff
@@ -0,0 +1,114 @@
...
+PyPI renders these URLs on the project page and splits them into `verified` and 
+`unverified` subgroups. They are also available using the 
+[JSON API](https://warehouse.pypa.io/api-reference/json.html).
+
+### Verified details
+
+![Verified details](assets/verified_details.png){ loading=lazy }
+
+PyPI currently supports several ways of verifying project URLs and highlights 
+them using a green checkmark (:fontawesome-solid-circle-check:{ .checked }).
```
Comment (woodruffw) [inline docs/user/project_metadata.md:26]: Nitpick, phrasing:

```suggestion
PyPI currently supports several ways of verifying project URLs. When a URL is verified, PyPI highlights it using a green checkmark (:fontawesome-solid-circle-check:{ .checked }).
```

**Review by woodruffw — APPROVED** [review]
LGTM, thanks @DarkaMaul!

**Comment by woodruffw** [comment]
> @woodruffw Want to give a final approval here?

Done, thanks!
